src/viam/sdk/common/client_helper.hpp

```diff
@@ -21,10 +38,7 @@ class ClientHelper {
     }
 
    public:
-    using PFn = ::grpc::Status (StubType::*)(::grpc::ClientContext*,
-                                             const RequestType&,
-                                             ResponseType*);
-    explicit ClientHelper(ClientType* client, StubType* stub, PFn pfn)
+    explicit ClientHelper(ClientType* client, StubType* stub, MethodType pfn)
```

Comment:
Is there some way to constrain `MethodType` to ensure that it matches `SyncMethodType` or `StreamingMethodType`?

Reply:
Actually, nevermind: the helper functions sort of already do that.

Reply:
I'm glad you asked! So yes the helper functions sort of already do it, I think we could constrain it further by defining the method typedefs as traits classes and then we could `static_assert` and/or define `ClientHelper` as a partial template specialization, but that might be getting a bit silly versus the "good enough" helper function approach.

Reply:
I think we go with "good enough" here. If you pass the wrong sort of thing it is unlikely to compile anyway. And if it *does* compile, well, then maybe that's OK too. Resolving.